librepo/downloader.c

```diff
@@ -1329,7 +1330,7 @@ check_finished_transfer_checksum(int fd,
                                               &calculated,
                                               err);
         if (!ret)
```

Comment:
IMHO, this won't work properly.

The ``ret`` which is tested here is a ``ret`` var defined in this context (context of the ``for`` loop during call of ``lr_checksum_fd_compare``). The newly defined ``ret`` in the main context of ``check_finished_transfer_checksum`` function will still be ``TRUE`` (it won't be set to FALSE) and after the control jump to the ``cleanup`` it will return ``TRUE`` and ``GError`` instead of ``FALSE`` and ``GError``.

Reply:
Indeed.  This should finally be fixed now, as per your instructions.